engine/runtime/src/main/java/org/enso/compiler/TreeToIr.java

```diff
@@ -1879,26 +1915,28 @@ def translateInvalid(invalid: AST.Invalid): Expression = {
     * Currently this only supports documentation comments, and not standarc
     * types of comments as they can't currently be represented.
     *
-    * @param tree the comment to transform
+    * @param doc the comment to transform
     * @return the [[IR]] representation of `comment`
     */
-  IR.Comment translateComment(Tree tree) {
-    return switch (tree) {
-      case Tree.ArgumentBlockApplication comment -> {
-        var doc = new StringBuilder();
-        var sep = "";
-        for (var l : comment.getArguments()) {
-          if (l.getExpression() != null) {
-            doc.append(sep);
-            doc.append(l.getExpression().codeRepr());
-            sep = "\n";
+  IR.Comment translateComment(Tree.Documented doc) {
```

Comment:
If this takes a `DocComment` argument, then you can use it for `TypeConstructorDef` and `Case` documentation fields.

Reply:
Attempted in [da5ad8f](https://github.com/enso-org/enso/pull/3806/commits/da5ad8f479feefdda0d6e46d877a25f9c40e2263), but not used yet. I guess I will need this in the future.